qa/integration-tests/src/test/java/io/camunda/it/rdbms/db/processinstance/ProcessInstanceIT.java

```diff
@@ -192,20 +169,75 @@ public void shouldFindProcessInstanceWithFullFilter(
+                                    .processDefinitionKeys(1337L)
+                                    .states(ProcessInstanceState.ACTIVE.name())
+                                    .parentProcessInstanceKeys(-1L)
+                                    .parentFlowNodeInstanceKeys(-1L))
+                        .sort(s -> s)
+                        .page(p -> p.from(0).size(5))));
 
     assertThat(searchResult.total()).isEqualTo(1);
-    assertThat(searchResult.hits()).hasSize(1);
-    assertThat(searchResult.hits().getFirst().key()).isEqualTo(processInstanceKey);
+    assertThat(searchResult.items()).hasSize(1);
+    assertThat(searchResult.items().getFirst().key()).isEqualTo(processInstanceKey);
+  }
+
+  @TestTemplate
+  public void shouldFindProcessInstanceWithSearchAfter(
+      final CamundaRdbmsTestApplication testApplication) {
+    final RdbmsService rdbmsService = testApplication.getRdbmsService();
+    final RdbmsWriter rdbmsWriter = rdbmsService.createWriter(PARTITION_ID);
+    final ProcessInstanceReader processInstanceReader = rdbmsService.getProcessInstanceReader();
+
+    final var processDefinition =
+        ProcessDefinitionFixtures.createAndSaveProcessDefinition(rdbmsWriter, b -> b);
+    createAndSaveRandomProcessInstances(
+        rdbmsWriter,
+        b ->
+            b.processDefinitionKey(processDefinition.processDefinitionKey())
+                .processDefinitionId(processDefinition.processDefinitionId()));
+    final var sort =
+        ProcessInstanceSort.of(
+            s ->
+                s.processDefinitionName()
+                    .asc()
+                    .processDefinitionVersion()
+                    .asc()
+                    .startDate()
+                    .desc());
+    final var searchResult =
+        processInstanceReader.search(
+            ProcessInstanceQuery.of(
+                b ->
+                    b.filter(f -> f.processDefinitionIds(processDefinition.processDefinitionId()))
+                        .sort(sort)
+                        .page(p -> p.from(0).size(20))));
+
+    final var instanceAfter = searchResult.items().get(9);
+    final var nextPage =
+        processInstanceReader.search(
+            ProcessInstanceQuery.of(
+                b ->
+                    b.filter(f -> f.processDefinitionIds(processDefinition.processDefinitionId()))
+                        .sort(sort)
+                        .page(
+                            p ->
+                                p.size(5)
+                                    .searchAfter(
+                                        new Object[] {
+                                          instanceAfter.processName(),
+                                          instanceAfter.processVersion(),
+                                          instanceAfter.startDate(),
+                                          instanceAfter.key()
+                                        }))));
+
+    assertThat(nextPage.total()).isEqualTo(20);
```

Comment:
🔧 It would be also nice to check that entities are also sorted properly. With different column types varchar, int, date etc.

Reply:
This happens in the `ProcessInstanceSortIT` (or other `{ENTITY}SortIT`'s).

Reply:
Ah okay, yes.`ProcessInstanceSortIT` tests the sorting and here we check the keyset results against the regular sorted results. Nice!